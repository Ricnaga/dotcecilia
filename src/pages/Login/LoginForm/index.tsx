import { CeciliaButton, CeciliaInput } from '../../../shared/components';
import { useLoginForm } from './useLoginForm';

export enum FormFields {
  name = 'name',
  password = 'password',
}

export function LoginForm() {
  const {
    data: { values },
    functions: { onChange, onSubmit },
  } = useLoginForm();

  return (
    <form onSubmit={onSubmit}>
      <div>
        <CeciliaInput
          placeholder="Digite o usuÃ¡rio"
          value={values[FormFields.name]}
          onChange={onChange(FormFields.name)}
        />
      </div>

      <div>
        <CeciliaInput
          placeholder="Digite a senha"
          type="password"
          value={values[FormFields.password]}
          onChange={onChange(FormFields.password)}
        />
      </div>

      <CeciliaButton title="Entrar" type="submit" />
    </form>
  );
}
