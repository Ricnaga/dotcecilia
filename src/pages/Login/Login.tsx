import { LoginForm } from './LoginForm/LoginForm';

export function Login() {
  return (
    <div className="grid grid-cols-1 max-w-xs mx-auto mt-32 p-4">
      <div className="text-center text-5xl my-8 whitespace-nowrap">
        <h1 className="font-bold">Fazer Login</h1>
      </div>
      <LoginForm />
    </div>
  );
}
