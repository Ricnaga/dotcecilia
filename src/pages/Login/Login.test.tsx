import { render } from '@testing-library/react';
import { Login } from './Login';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useNavigate: jest.fn(),
}));

describe('Page: Login', () => {
  it('should render correctly', () => {
    const { container } = render(<Login />);
    expect(container).toMatchSnapshot();
  });
});
