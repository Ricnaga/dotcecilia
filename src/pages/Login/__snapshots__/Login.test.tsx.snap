// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Login should render correctly 1`] = `
<div>
  <div
    class="grid grid-cols-1 max-w-xs mx-auto mt-32 p-4"
  >
    <div
      class="text-center text-5xl my-8 whitespace-nowrap"
    >
      <h1
        class="font-bold"
      >
        Fazer Login
      </h1>
    </div>
    <form>
      <div
        class="mb-4"
      >
        <div>
          <div
            data-error="false"
            data-variants="input-wrapper"
          >
            <input
              data-variants="input"
              placeholder="Digite o usuÃ¡rio"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="mb-4"
      >
        <div>
          <div
            data-error="false"
            data-variants="input-wrapper"
          >
            <input
              data-variants="input"
              placeholder="Digite a senha"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="w-full mt-2"
        data-variants="curtain"
        type="submit"
      >
        <span
          data-text="default"
        >
          Entrar
        </span>
      </button>
    </form>
  </div>
</div>
`;
